### PR TITLE
IDE-678 ESDL Publish and Generate ECL fail from nested folder in local repository

### PR DESCRIPTION
**Pull request by dehilsterlexis**
Fixed IDE-678

Signed-off-by: dehilsterlexis <david.dehilster@lexisnexis.com>

### CODE REVIEW

**Comment by dehilsterlexis** [comment]
@GordonSmith for review

File: comms/ModFileRepository.cpp [hunk 1]
```diff
@@ -120,10 +120,15 @@ class CModFileRepository : public IRepository, public clib::CLockableUnknown
...
+        typedef std::vector<std::_tstring> split_vector_type;
         for(IModuleVector::iterator itr = allModules.begin(); itr != allModules.end(); ++itr)
         {
             IModule * parent = itr->get()->GetParentModule();
-            if ((!parent && module.empty()) || (parent && boost::algorithm::equals(module, parent->GetQualifiedLabel())))
+            std::_tstring label = itr->get()->GetQualifiedLabel();
+            split_vector_type lableParts;
+            boost::algorithm::split(lableParts, label, boost::algorithm::is_any_of(CModuleHelper::DELIMTER), boost::algorithm::token_compress_on);
+
+            if ((parent && lableParts.size() > 1) || (!parent && module.empty()) || (parent && boost::algorithm::equals(module, parent->GetQualifiedLabel())))
```
Comment (GordonSmith) [inline comms/ModFileRepository.cpp]: This only checks up 1 level - don't we need to walk the entire "labelParts" and each level to the module list?

Comment (dehilsterlexis) [reply comms/ModFileRepository.cpp]: @GordonSmith I will check this more thoroughly.

**Comment by dehilsterlexis** [comment]
@GordonSmith back to you. Added all the folders as modules.

**Comment by GordonSmith** [comment]
Looks ok (have you tested with nested modules greater than a depth of 2?)

**Comment by dehilsterlexis** [comment]
@GordonSmith yes. I tried three levels and they all work.

**Pull request merged**